Add hash and equality operators for WorkerInfo

aazzolini opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#29958 Add hash and equality operators for WorkerInfo**

DistributedOptimizer relies on hashing WorkerInfo in order to coalesce fan-out RPCs. This will likely be a very common use case (EASGD will do the same, for example).

Differential Revision: [D18548257](https://our.internmc.facebook.com/intern/diff/D18548257/)

aazzolini committed:
Add hash and equality operators for WorkerInfo

DistributedOptimizer relies on hashing WorkerInfo in order to coalesce fan-out RPCs. This will likely be a very common use case (EASGD will do the same, for example).

Differential Revision: [D18548257](https://our.internmc.facebook.com/intern/diff/D18548257/)

[ghstack-poisoned]

pritamdamania87 reviewed (approved):
Thanks for the quick fix!

aazzolini committed:
Update on "Add hash and equality operators for WorkerInfo"


DistributedOptimizer relies on hashing WorkerInfo in order to coalesce fan-out RPCs. This will likely be a very common use case (EASGD will do the same, for example).

Differential Revision: [D18548257](https://our.internmc.facebook.com/intern/diff/D18548257/)

[ghstack-poisoned]

aazzolini committed:
Update on "Add hash and equality operators for WorkerInfo"


DistributedOptimizer relies on hashing WorkerInfo in order to coalesce fan-out RPCs. This will likely be a very common use case (EASGD will do the same, for example).

Differential Revision: [D18548257](https://our.internmc.facebook.com/intern/diff/D18548257/)

[ghstack-poisoned]
